packages/twenty-front/src/modules/settings/page-layout/utils/__tests__/createUpdatedTabLayouts.test.ts

```diff
@@ -0,0 +1,82 @@
+import { type TabLayouts } from '../../states/pageLayoutCurrentLayoutsState';
+import { createUpdatedTabLayouts } from '../createUpdatedTabLayouts';
+
+describe('createUpdatedTabLayouts', () => {
+  const mockTabLayouts: TabLayouts = {
+    'tab-1': {
+      desktop: [{ i: 'widget-1', x: 0, y: 0, w: 2, h: 2 }],
+      mobile: [{ i: 'widget-1', x: 0, y: 0, w: 1, h: 2 }],
+    },
+    'tab-2': {
+      desktop: [],
+      mobile: [],
+    },
+  };
+
+  const newLayout = { i: 'widget-2', x: 2, y: 0, w: 3, h: 3 };
+
+  it('should add new layout to existing tab', () => {
+    const result = createUpdatedTabLayouts(mockTabLayouts, 'tab-1', newLayout);
+
+    expect(result['tab-1'].desktop).toHaveLength(2);
```

Comment:
I feel using snapshot testing could be a good usecase here but Im' fine with the current tests